components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/utils/ServiceCatalogUtils.java

```diff
@@ -871,4 +894,17 @@ private static String getProxyServiceUrlFromMetadata(File yamlFile) {
         }
         return currentServiceUrl;
     }
+
+    public static boolean isServiceCatalogEnabled() {
```

Comment:
we are doing this check for each carbon app get hot deployed. 
shall we do the check once and save the flag in cappDeployer

Reply:
Fixed.